azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/LoadBalancerImpl.java

```diff
@@ -171,22 +184,29 @@ protected void beforeCreating()  {
 
     @Override
     protected void afterCreating() {
+        List<Exception> nicExceptions = new ArrayList<>();
+
         // Update the NICs to point to the backend pool
         for (Entry<String, String> nicInBackend : this.nicsInBackends.entrySet()) {
             String nicId = nicInBackend.getKey();
             String backendName = nicInBackend.getValue();
+            NetworkInterface nic = this.manager().networkInterfaces().getById(nicId);
+            NicIpConfiguration nicIp = nic.primaryIpConfiguration();
             try {
-                NetworkInterface nic = this.manager().networkInterfaces().getById(nicId);
-                NicIpConfiguration nicIp = nic.primaryIpConfiguration();
                 nic.update()
                     .updateIpConfiguration(nicIp.name())
-                        .withExistingLoadBalancerBackend(this, backendName)
-                        .parent()
+                    .withExistingLoadBalancerBackend(this, backendName)
+                    .parent()
                     .apply();
             } catch (Exception e) {
-                e.printStackTrace();
+                nicExceptions.add(e);
             }
         }
```

Comment:
I mean this:

We `[1] fetch` and `[2] update` a list of NICs i.e. `processing` each NIC involves two API calls. But we aggregate only  the exception happens during `[2] update` of each NIC and finally throw it. In this case, shouldn't we include `[1] fetch` also in the try block ?